### PR TITLE
(#5154) - Add deprecation notice for db.put(doc, id, rev)

### PR DESCRIPTION
**Pull request by daleharvey**

### CODE REVIEW

File: src/adapter.js [hunk 1]
```diff
@@ -218,6 +218,9 @@ AbstractPouchDB.prototype.put =
     temp = args.shift();
     temptype = typeof temp;
     if (temptype === "string" && !id) {
+      console.warn('db.put(doc, id, rev) has been deprecated and will be ' +
+                   'removed in a future release, please use ' +
+                   'db.put({_id: id, _rev: rev}) instead');
       doc._id = temp;
       id = true;
     } else if (temptype === "string" && id && !('_rev' in doc)) {
```
Comment (nolanlawson) [inline src/adapter.js]: I think we also need to put a console warning here. I would like to deprecate everything but `put(doc)` or `put(doc, callback)`.

Comment (daleharvey) [inline src/adapter.js]: hrm yeh, may need another way to do this notice, as far as I can tell from the code putting another warn there would double warn for `db.put(doc, id, rev)`

Comment (nolanlawson) [inline src/adapter.js]: Double-warning is fine in my opinion; really yell at 'em for using an old API! :D

**Comment by daleharvey** [comment]
Updated, I would still prefer it to only print out once, not so worried about the code being ugly since we can delete it all shortly

**Comment by nolanlawson** [comment]
OK, I'm fine with that. The important thing is that it's deprecated in the API at least.

**Comment by daleharvey** [comment]
Ah yeh forgot to add back to the docs when I rebased, done that, cheers

**Pull request merged**